mock/mock.go

```diff
@@ -102,15 +108,38 @@ func (c *Call) unlock() {
 	c.Parent.mutex.Unlock()
 }
 
+// If the only return arg is a function which takes and returns Arguments, invoke it instead of returning it as the value
+func (c *Call) getReturnArguments(args Arguments) Arguments {
+	if c.returnFunc != nil {
+		return c.returnFunc(args)
+	}
+
+	return c.ReturnArguments
+}
+
 // Return specifies the return arguments for the expectation.
 //
 //	Mock.On("DoSomething").Return(errors.New("failed"))
+//
+// If you pass a single returnArg which is a function that itself takes Arguments and returns Arguments,
```

Comment:
This should use different method. `Mock` is already not trivial to understand, each method should strive to do one thing and do it well.